easybuild/framework/easyblock.py

```diff
@@ -1444,8 +1443,19 @@ def extensions_step(self, fetch=False):
         self.clean_up_fake_module(fake_mod_data)
 
     def package_step(self):
-        """Package software (e.g. into an RPM)."""
-        pass
+        """Prepare package software (e.g. into an RPM) with fpm."""
+        rpmname = "HPCBIOS.20150211-%s-%s" % (self.name, self.version)
+        os.chdir(os.environ['TMPDIR'])
+        
+        if self.toolchain.name == "dummy":
+            dependencies = []
+        else:
+            dependencies = [ "=".join([ self.toolchain.name, self.toolchain.version ]) ]
+        dependencies.extend([ "=".join([ dep['name'], dep['version'] ]) for dep in self.cfg.dependencies() ])
+        depstring = '--dependency ' + ' --dependency '.join(dependencies)
+
+        cmd = "fpm --workdir $TMPDIR -t rpm --name %s -s dir %s -C %s" % (rpmname, depstring, self.installdir)
```

Comment:
Hope you don't mind me dropping in on the review!

When invoking fpm (or subprocesses, in general), I highly recommend using array-oriented execution instead of one string. This allows you to avoid any problems that might occur with spaces in the filename or other similar problems.

I had a look at the code under `run_cmd` and it uses `subprocess.Popen` which takes either a string or a list/tuple as program args, I think you can do this like:

```python
cmd = ("fpm", "--workdir", os.environ["TMPDIR"], "-t", "rpm", "--name", rpmname, "-s", "dir") + dependency_flags + ("-C", self.installdir)
```

The above assumes 'dependency_flags' is a tuple of flags, not a string.

This will help you if depstring or installdir or $TMPDIR have spaces or other possible hazards. Hope this helps! :)